### DISCUSSION TITLE
upsert or insert...

### DISCUSSION THREAD

**Discussion by jkohlin** [supabase/supabase#10314] [Questions] [answered]
On Discord I was told not to use upsert(data) but instead insert(data, {upsert:true}).
However, looking at the docs for v.2 upsert is still in the list of functions for database queries.
So which one is it?

...and a follow up question: 
Can you get the upserted records as the results?

**Comment by GaryAustin1** [comment, marked as answer]
v2 javascript-js has removed .insert() with upsert:true and now you must use .upsert().

I'm not sure what you are asking on your 2nd question, but adding .select() to the end of the upsert will return data.

**Reply by jkohlin** [reply]
I see, thanks.
For the second question, will select() return only the rows that were actually updated?

**Reply by GaryAustin1** [reply]
It should only return changed rows (versus all rows in the table) if that is what you are asking.  But I've not actually used it.
